compiler/sempass2.nim

```diff
@@ -721,6 +721,17 @@ proc cstringCheck(tracked: PEffects; n: PNode) =
     message(tracked.config, n.info, warnUnsafeCode, renderTree(n))
 
 proc track(tracked: PEffects, n: PNode) =
+  template gcsafeAndSideeffectCheck() =
```

Comment:
What about some small refactoring such as?

```suggestion
  template gcsafeAndSideeffectCheck() =
    if not importedFromC(a) and not (a.kind == nkSym and a.sym == tracked.owner):
      if notGcSafe(op):
        if warnGcUnsafe in tracked.config.notes: warnAboutGcUnsafe(n, tracked.config)
        markGcUnsafe(tracked, a)
      if tfNoSideEffect notin op.flags:
        markSideEffect(tracked, a)
```

Reply:
Better check for the uncommon cases first though. (Cargo cult approach to "efficiency".)